wasm/wasm_source/src/vp_masp.rs

```diff
@@ -70,9 +137,50 @@ fn validate_tx(
         }
 
         // Handle unshielding/transparent output
+        // The following boundary conditions must be satisfied
+        // 1. Zero transparent inupt
+        // 2. One transparent output
+        // 3. Asset type must be properly derived
+        // 4. Value from the output must be the same as the containing transfer
+        // 5. Public key must be the hash of the target
         if transfer.target != masp() {
+            // Satisfies 1.
+            if !shielded_tx.vin.is_empty() {
```

Comment:
This check enforces that transactions with non-MASP targets mut not have any transparent inputs. While this check can be done, the transaction would still be valid (i.e. leaving the shielded pool and transparent pool in a consistent state) without it. Perhaps we can remove this check to simplify the logic?